cloudwatch/cloudwatch.go

```diff
@@ -272,25 +272,36 @@ func (output *OutputPlugin) cleanUpExpiredLogStreams() {
 	}
 }
 
-func (output *OutputPlugin) getLogStream(tag string) (*logStream, error) {
+func (output *OutputPlugin) getLogStream(tag, groupName string) (*logStream, error) {
 	// find log stream by tag
 	name := output.getStreamName(tag)
 	stream, ok := output.streams[name]
-	if !ok {
-		// stream doesn't exist, create it
-		stream, err := output.createStream(tag)
+	if ok {
+		return stream, nil
+	}
+	// extra check, empty groupName indicates flush phase, but still land here meaning unable to find stream
+	if groupName == "" {
+		return nil, fmt.Errorf("error: does not compute: Log Stream %s could not be created, but also could not be found in the log group", name)
```

Comment:
This error message is copied from elsewhere, if there was ever a bug in this code it's more useful if it uses different text. May be something like "does not compute: in call to Flush() but log stream not found"